templates/install.j2

```diff
@@ -71,40 +59,117 @@ ynh_setup_source --dest_dir="$install_dir"
 {% endif %}
 chown -R $app:www-data "$install_dir"
 
+#=================================================
+# ADD A CONFIGURATION
+#=================================================
+ynh_script_progression "Adding app's configuration file..."
+
+{% if data.generator_mode == 'tutorial' -%}
+### You can add specific configuration files.
+###
+### Typically, put your template conf file in ../conf/your_config_file
+### The template may contain strings such as __FOO__ or __FOO_BAR__,
+### which will automatically be replaced by the values of $foo and $foo_bar
+###
+### ynh_config_add will also keep track of the config file's checksum,
+### which later during upgrade may allow to automatically backup the config file
+### if it's found that the file was manually modified
+###
+### Check the documentation of `ynh_config_add` for more info.
+{% endif %}
+
+ynh_config_add --template="{{ data.custom_config_file }}" --destination="$install_dir/{{ data.custom_config_file }}"
+
+{% if data.generator_mode == 'tutorial' -%}
+# FIXME: this should be handled by the core in the future
+### You may need to use chmod 600 instead of 400,
+### for example if the app is expected to be able to modify its own config
+{% endif %}
+chmod 400 "$install_dir/{{ data.custom_config_file }}"
+chown "$app:$app" "$install_dir/{{ data.custom_config_file }}"
+
+{% if data.generator_mode == 'tutorial' -%}
+### For more complex cases where you want to replace stuff using regexes,
+### you shoud rely on ynh_replace_string (which is basically a wrapper for sed)
+### When doing so, you also need to manually call ynh_store_file_checksum
+###
+### ynh_replace_string --match_string="match_string" --replace_string="replace_string" --target_file="$install_dir/{{ data.custom_config_file }}"
+### ynh_store_file_checksum --file="$install_dir/{{ data.custom_config_file }}"
+{% endif %}
+
 #=================================================
 # SYSTEM CONFIGURATION
 #=================================================
-ynh_script_progression --message="Adding system configurations related to $app..."
+ynh_script_progression "Adding system configurations related to $app..."
+
+{% if data.generator_mode == 'tutorial' -%}
+### `ynh_config_add_phpfpm` is used to set up a PHP config.
+### You can remove it if your app doesn't use PHP.
+### `ynh_config_add_phpfpm` will use the files conf/extra_php-fpm.conf
+### If you're not using these lines:
+###   - You can remove these files in conf/.
+###   - Remove the section "BACKUP THE PHP-FPM CONFIGURATION" in the backup script
+###   - Remove also the section "REMOVE PHP-FPM CONFIGURATION" in the remove script
+###   - As well as the section "RESTORE THE PHP-FPM CONFIGURATION" in the restore script
+###     with the reload at the end of the script.
+###   - And the section "PHP-FPM CONFIGURATION" in the upgrade script
+{% endif %}
```

Comment:
Hmmm but this whole block is not really relevant because all the stuff of "remove the section from the backup script" etc is handled automagically x_X